### PR TITLE
NIAD-2846: new RCMR_IN030000UK07 compatibility test

### PR DESCRIPTION
**Pull request by ole4ryb**
## What

Adding an additional test to make sure that existing codebase can handle RCMR_IN030000UK07 xml type

## Why

The current implementation works with RCMR_IN030000UK06 xml type. The change adds a new test to make sure that the current codebase can handle RCMR_IN030000UK07 type without any errors.     

## Checklist:

- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have updated the [Changelog](CHANGELOG.md) with details of my change in the UNRELEASED section if this change will affect end users

### CODE REVIEW

File: gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java [hunk 1]
```diff
@@ -107,10 +110,36 @@ public void setUp() {
     @Test
     public void handleEhrExtractFromQueue() throws JSONException {
         // process starts with consuming a message from MHS queue
-        sendInboundMessageToQueue("/xml/RCMR_IN030000UK06/payload_part.xml");
+        sendInboundMessageToQueue("/xml/RCMR_IN030000UK06/payload_part.xml", EBXML_PART_PATH);
+
+        // wait until EHR extract is translated to bundle resource and saved to the DB
+        waitAtMost(Duration.ofSeconds(WAITING_TIME)).until(this::isEhrMigrationCompleted);
```
Comment (adrianclay) [inline gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java:116]: What was the thinking behind specifying the Duration here? I'd say that if the integration test doesn't complete within 10 seconds that's very slow, and should be a failure. 30 seconds is unbelievable slow.

File: gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java [hunk 2]
```diff
@@ -107,10 +110,36 @@ public void setUp() {
...
+
+        // verify generated bundle resource
+        verifyBundle("/json/expectedBundle.json");
+    }
+
+    @Test
+    public void handleEhrExtractWithConfidentialitySchemaFromQueue() throws JSONException {
```
Comment (adrianclay) [inline gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java:123]: This and the test below have very similar names, not something that needs fixing immediately, but each test should be clear about what it's testing.

File: gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java (hunk 1 above)
Comment (ole4ryb) [reply gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java:116]: I think we can change it back to the original 10 sec, it should be fine.

File: gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java (hunk 2 above)
Comment (ole4ryb) [reply gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/EhrExtractHandlingIT.java:123]: yes, agree, I can take a note of it and do it at the end of all the refactoring

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/ObservationUtil.java [hunk 3]
```diff
@@ -76,10 +76,11 @@ public static CodeableConcept getInterpretation(CV interpretationCode) {
     }
 
     public static List<Observation.ObservationReferenceRangeComponent> getReferenceRange(
-        List<RCMRMT030101UK04ReferenceRange> referenceRangeList) {
+        List<? extends RCMRMT030101UKReferenceRange> referenceRangeList) {
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/ObservationUtil.java:79]: I'm no Java expert, what does the `? extends` provide here?

File: schema/src/test/resources/xml/RCMR_IN030000UK07.xml
Comment (adrianclay) [inline schema/src/test/resources/xml/RCMR_IN030000UK07.xml:1]: Is this file being used at all?

File: schema/src/main/java/org/hl7/v3/RCMRMT030101UKAuthor2.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package org.hl7.v3;
+
+public interface RCMRMT030101UKAuthor2 {
```
Comment (adrianclay) [inline schema/src/main/java/org/hl7/v3/RCMRMT030101UKAuthor2.java:3]: I'm so confused by this interface 🤣

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/BundleMapperService.java [hunk 5]
```diff
@@ -243,12 +245,30 @@ private Organization getPatientOrganization(List<? extends DomainResource> agent
             .orElse(null);
     }
 
-    private RCMRMT030101UK04EhrFolder getEhrFolder(RCMRIN030000UK06Message xmlMessage) {
-        return xmlMessage.getControlActEvent().getSubject().getEhrExtract().getComponent().get(0).getEhrFolder();
+    private RCMRMT030101UK04EhrFolder getEhrFolder(RCMRIN030000UKMessage xmlMessage) {
+        if (xmlMessage instanceof RCMRIN030000UK07Message) {
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/BundleMapperService.java:249]: What's the plan for this `if` statements (and others), they look a bit grim are them temprary?

**Review by adrianclay — APPROVED** [review]

**Pull request merged**